mlir/utils/jenkins/Jenkinsfile

```diff
@@ -1057,5 +1058,68 @@ pipeline {
         }
+
+        stage ("Code coverage") {
+            when {
+                beforeAgent true;
+                equals expected: true, actual: params.runCodeCoverage;
+            }
+            matrix {
+                axes {
+                    axis {
+                        name 'CHIP'
+                        values 'gfx908', 'gfx90a'    //, 'gfx1030', 'gfx1100'
+                    }
+                }
+                agent {
+                    docker {
+                        image dockerImage()
+                        args dockerArgs()
+                        label getLabelFromChip("${CHIP}")
+                        alwaysPull true
+                    }
+                }
+                environment {
+                    PATH="/opt/rocm/llvm/bin:$PATH"
+                    HOME="${WORKSPACE}"
+                    // Note the %m to avoid issues with threads and dynamic libraries.
+                    LLVM_PROFILE_FILE="${WORKSPACE}/build/%m-%p.profraw"
+                }
+                stages {
+                    stage ("body") {
+                        steps {
+                            // Build with profiling on, and just code-generation tests.
+                            sh 'rm -f build/CMakeCache.txt'
+                            sh 'rm -f build/*.profraw'
+                            buildProject('check-rocmlir-build-only',
+                                         '-DBUILD_FAT_LIBROCKCOMPILER=ON -DCMAKE_BUILD_TYPE=debug -DCMAKE_CXX_FLAGS_DEBUG="-O0 -fprofile-instr-generate -fcoverage-mapping"')
+                            dir ('build') {
+                                // Run tests.
+                                sh 'ninja check-rocmlir'
+                                // Profile processing.
+                                sh '/opt/rocm/llvm/bin/llvm-profdata merge -sparse ./*.profraw -o ./coverage.profdata'
+                                sh '/opt/rocm/llvm/bin/llvm-cov report --object ./bin/rocmlir-opt --object ./bin/rocmlir-driver --object ./bin/rocmlir-gen --instr-profile ./coverage.profdata --ignore-filename-regex=external/llvm-project > ./coverage.report'
+                                sh 'cat ./coverage.report'
+                                sh '/opt/rocm/llvm/bin/llvm-cov show --object ./bin/rocmlir-opt --object ./bin/rocmlir-driver --object ./bin/rocmlir-gen --instr-profile ./coverage.profdata --ignore-filename-regex=external/llvm-project -Xdemangler=/opt/rocm/llvm/bin/llvm-cxxfilt > ./coverage.info'
+                                sh '/opt/rocm/llvm/bin/llvm-cov show --object ./bin/rocmlir-opt --object ./bin/rocmlir-driver --object ./bin/rocmlir-gen --instr-profile ./coverage.profdata --ignore-filename-regex=external/llvm-project -Xdemangler=/opt/rocm/llvm/bin/llvm-cxxfilt --format=html > ./coverage.html'
+                                // Upload to codecov.
+                                withCredentials([string(credentialsId: 'codecov-token-rocmlir', variable: 'CODECOV_TOKEN')]) {
```

Comment:
The token can be found at https://app.codecov.io/gh/ROCmSoftwarePlatform/rocMLIR/settings and I've added it to rocmhead.amd.com:8080's credentials.